cms-patatrack/cmssw
# GPU2CPU for clusters and RecHIts

to test it add to your configuration
```
process.siPixelDigis = cms.EDProducer("SiPixelRawToDigiGPU",
    CablingMapLabel = cms.string(''),
    ConvertADCtoElectrons = cms.bool(False),
    ErrorList = cms.vint32(29),
    IncludeErrors = cms.bool(True),
    InputLabel = cms.InputTag("rawDataCollector"),
    Regions = cms.PSet(

    ),
    Timing = cms.untracked.bool(False),
    UsePhase1 = cms.bool(True),
    UsePilotBlade = cms.bool(False),
    UseQualityInfo = cms.bool(False),
    UserErrorList = cms.vint32(40)
)

process.siPixelRecHitsPreSplitting = cms.EDProducer("SiPixelRecHitGPU",
    src = cms.InputTag("siPixelDigis"),  
    CPE = cms.string('PixelCPEFast'),
    VerboseLevel = cms.untracked.int32(0),
)

from FWCore.ParameterSet.MassReplace import massReplaceInputTag
massReplaceInputTag(process, 'siPixelClustersPreSplitting', 'siPixelDigis')

```

to ensure fair comparison, on CPU add
```
process.siPixelClustersPreSplitting.payloadType = 'HLT'
```

for data remember to change GT to
```
process.GlobalTag = GlobalTag(process.GlobalTag, '100X_dataRun2_asv2plusPixelGainfromHLT_v1', '')
```

printouts and asserts can be cleaned up later on
